.github/workflows/allow-merge.yml

```diff
@@ -0,0 +1,32 @@
+# This workflow will just make sure that the PR cannot be merged during release by check if there is a release PR opened.
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Allow Merge
+
+on:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  can-merge:
+    name: Check If Can Merge
+    if: ${{ github.head_ref != 'release' }}
```

Comment:
I believe this check is for running this job for all non release PRs. In case of release PR's the job will be skipped and there should be no impediments for release?